File: arrays/string_matching_in_an_array.rb
# @param {String[]} words
# @return {String[]}
def string_matching(words)
    results = []
    
    words.each do |search_word|
        words.each do |match_word|
        next if search_word == match_word
            found = search_word.scan(match_word)
            if found.any?
                results << found
            end
        end
    end
    results.flatten.uniq
end

File: arrays/string_matching_in_an_array.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        results = []
    
        for search_word in words:
            for match_word in words:
                if search_word == match_word:
                    continue
                if match_word in search_word:
                    results.append(match_word)
        
        return list(set(results))